lib/MusicBrainz/Server/Controller/Collection.pm

```diff
@@ -70,6 +70,11 @@ sub _do_add_or_remove {
     if ($entity_id) {
         my $entity = $c->model(type_to_model($entity_type))->get_by_id($entity_id);
 
+        unless ($entity) {
+            $c->stash->{message} = 'This ID does not match any entity; the entity might have been merged or deleted';
```

Comment:
It should use the same message (either the current message or the new one) for [`invalid_mbid`](https://github.com/metabrainz/musicbrainz-server/blob/v-2022-03-28/lib/MusicBrainz/Server/Controller.pm#L33) or be more specific; for example: _No entity has been added to the collection, because this ID does not match any entity._ (Removing an already deleted entity from a collection can probably be silent.)

Reply:
I don't think a delete should be silent, because the user might have just incorrectly entered the ID and want to be notified (for example if they're doing this through a bot or something).

This is a row id, not an MBID, which is why I didn't reuse the invalid MBID message, but we could be more precise with the message :)

Reply:
I wouldn't mind if it was phrased similarly to `invalid_mbid`, like `'{id}' is not a valid row ID; the entity might have been merged or deleted`.  Though for `invalid_mbid`, invalid there means "not a UUID," whereas here the message could be triggered by either a "not an ID" situation (like `0` or `abc123`), or a 404-type situation (merged or deleted), from what I can tell, so the explanation at the end makes sense to me.